redis/parser/parser.go

```diff
@@ -76,6 +76,7 @@ func parse0(reader io.Reader, ch chan<- *Payload) {
 	defer func() {
 		if err := recover(); err != nil {
 			logger.Error(err, string(debug.Stack()))
+			ch <- nil
```

Comment:
close(ch) may be better

Reply:
yes, agree with you~ I don't know if it's useful for anything else, so I used "nil", i will fix it and re-PR